python/lib/communication/dmod/communication/evaluation_request.py

```diff
@@ -0,0 +1,163 @@
+import typing
+import abc
+import json
+
+from numbers import Number
+from typing import Dict
+from typing import Union
+
+from .maas_request import MessageEventType, Response
+from .message import Message
+
+SERIALIZABLE_DICT = typing.Dict[str, typing.Union[str, Number, dict, typing.List]]
+
+
+class EvaluationRequest(Message, abc.ABC):
+    """
+    A request to be forwarded to the evaluation service
+    """
+
+    event_type: MessageEventType = MessageEventType.EVALUATION_REQUEST
+    """ :class:`MessageEventType`: the event type for this message implementation """
+
+    @classmethod
+    @abc.abstractmethod
+    def get_action(cls) -> str:
+        ...
+
+    @property
+    def action(self) -> str:
+        return self.get_action()
+
+
+class EvaluationConnectionRequest(EvaluationRequest):
+    """
+    A request used to communicate through a chained websocket connection
+    """
+    _action_parameters: typing.Dict[str, typing.Any]
+
+    def __init__(self, **kwargs):
+        self._action_parameters = kwargs or dict()
+
+    @classmethod
+    def get_action(cls) -> str:
+        return "connect"
+
+    @property
+    def parameters(self) -> typing.Dict[str, typing.Any]:
+        return self._action_parameters
+
+    @classmethod
+    def factory_init_from_deserialized_json(cls, json_obj: dict):
```

Comment:
That threw me off, too: the deserializer basic rolls through a list of approved request classes that all have this function. If it successfully deserializes, it is interpreted as being the type of class to use, which then indicates the type of handler to use, so that's just part of the infrastructure at this point in the request service. I'll go ahead and add the annotation.